src/ui/public/state_management/state_storage/hashing_store.js

```diff
@@ -50,23 +48,13 @@ export default class HashingStore {
    *  @param {any} the value to hash
    *  @return {string} the hash of the value
    */
-  add(object) {
+  hashAndSetItem(object) {
```

Comment:
I think the actions of `hashAndSetItem` and `getItemAtHash` are a bit more complementary than `add` and `lookup`, especially since `add` doesn't imply you're getting a hash back, which seems like a really critical feature here.